cli/src/init.rs

```diff
@@ -0,0 +1,242 @@
+
+pub fn init_project(
+    arg_adapter: &Option<String>,
+    arg_project_name: &Option<String>,
+) -> Result<(), Box<dyn Error>> {
+    let project_name: String = match arg_project_name.as_deref() {
+        None => Input::with_theme(&ColorfulTheme::default())
+            .with_prompt("Project name")
+            .with_initial_text("mopro-example-app".to_string())
+            .interact_text()?,
+        Some(name) => name.to_string(),
+    };
+
+    let adapters = vec!["circom", "halo2"];
+
+    let selection = match arg_adapter.as_deref() {
+        None => MultiSelect::with_theme(&ColorfulTheme::default())
+            .with_prompt("Pick the adapters you want to use (multiple selection with space)")
+            .items(&adapters)
+            .interact()
+            .unwrap(),
+        Some(a) => {
+            let mut selection = vec![];
+            for (i, adapter) in adapters.iter().enumerate() {
+                if a.contains(adapter) {
+                    selection.push(i);
+                }
+            }
+            selection
+        }
+    };
+
+    if selection.is_empty() {
+        style::print_yellow("No adapters selected. Use space to select an adapter".to_string());
+        init_project(arg_adapter, &Some(project_name))?;
+    } else {
+        let current_dir = env::current_dir()?;
+
+        let project_dir = current_dir.join(&project_name);
+        fs::create_dir(&project_dir)?;
+
+        // Change directory to the project directory
+        env::set_current_dir(&project_dir)?;
+        const TEMPLATE_DIR: Dir = include_dir!("$CARGO_MANIFEST_DIR/src/template/init");
+
+        copy_embedded_dir(&TEMPLATE_DIR, &project_dir, selection.clone())?;
+
+        if let Some(cargo_toml_path) = project_dir.join("Cargo.toml").to_str() {
+            replace_project_name(cargo_toml_path, &project_name)?;
+            replace_features(
+                cargo_toml_path,
+                selection.iter().map(|&i| adapters[i]).collect(),
+            )?;
+            if selection.contains(&0) {
+                // circom is selected
+                circom_dependencies_template(cargo_toml_path)?;
+            }
+            if selection.contains(&1) {
+                // halo2 is selected
+                halo2_dependencies_template(cargo_toml_path)?;
+            }
+        }
+
+        if let Some(build_rs_path) = project_dir.join("build.rs").to_str() {
+            if selection.contains(&0) {
+                circom_build_template(build_rs_path)?;
+            }
+        }
+
+        if let Some(lib_rs_path) = project_dir.join("src").join("lib.rs").to_str() {
+            if selection.contains(&0) {
+                // circom is selected
+                circom_lib_template(lib_rs_path)?;
+            }
+            if selection.contains(&1) {
+                // halo2 is selected
+                halo2_lib_template(lib_rs_path)?;
+            }
+        }
+
+        // Print out the instructions
+        println!(
```

Comment:
This isn't a necessary change, but you can do multiline strings by default. e.g.

```rust
let hello = "hello
world
more text
";
println!("{hello}");
```